specification/2.0/README.md

```diff
@@ -3903,67 +3891,234 @@ Application-specific data.
-<br>
-`const black = rgb(0, 0, 0)`
+Introducing the visibility function
 
-*c<sub>diff</sub>* = `lerp(baseColor.rgb * (1 - dielectricSpecular.r), black, metallic)`
-<br>
-*F<sub>0</sub>* = `lerp(dieletricSpecular, baseColor.rgb, metallic)`
-<br>
-*&alpha;* = `roughness ^ 2`
+<img src="https://render.githubusercontent.com/render/math?math=\displaystyle V = \frac{G}{4 \, \left| N \cdot L \right| \, \left| N \cdot V \right|}">
 
-Additionally,  
-*V* is the normalized vector from the shading location to the eye  
-*L* is the normalized vector from the shading location to the light  
-*N* is the surface normal in the same space as the above values  
-*H* is the half vector, where *H* = normalize(*L*+*V*)  
+simplifies the original microfacet BRDF to
 
-The core lighting equation the sample uses is the Schlick BRDF model from [An Inexpensive BRDF Model for Physically-based Rendering](https://www.cs.virginia.edu/~jdl/bib/appearance/analytic%20models/schlick94b.pdf)
+<img src="https://render.githubusercontent.com/render/math?math=\displaystyle \text{MicrofacetBRDF} = V D">
 
-![](figures/lightingSum.PNG)
+with
 
-Below are common implementations for the various terms found in the lighting equation.
+<img src="https://render.githubusercontent.com/render/math?math=\displaystyle V = \frac{\, \chi^%2B(H \cdot L)}{\left| N \cdot L\right| %2B \sqrt{\alpha^2 %2B (1 - \alpha^2) (N \cdot L)^2}} \frac{\, \chi^%2B(H \cdot V)}{\left| N \cdot V \right| %2B \sqrt{\alpha^2 %2B (1 - \alpha^2) (N \cdot V)^2}}">.
 
-### Surface Reflection Ratio (F)
+Thus we have the function
 
-**Fresnel Schlick**
+```
+function specular_brdf(α) {
+  return V * D
+}
+```
+
+### Diffuse BRDF
+
+The diffuse reflection `diffuse_brdf(color)` is a Lambertian BRDF
+
+<img src="https://render.githubusercontent.com/render/math?math=\displaystyle \text{LambertianBRDF} = \frac{1}{\pi}">
+
+multiplied with the `color`.
+
+```
+function diffuse_brdf(color) {
+  return (1/pi) * color
+}
+```
+
+### Fresnel
+
+An inexpensive approximation for the Fresnel term that can be used for conductors and dielectrics was developed by [Schlick (1994)](#Schlick1994):
+
+<img src="https://render.githubusercontent.com/render/math?math=\displaystyle F = f_0 %2B (1 - f_0) (1 - \left| V \cdot H \right| )^5">
+
+The conductor Fresnel `conductor_fresnel(f0, bsdf)` applies a view-dependent tint to a BSDF:
+
+```
+function conductor_fresnel(f0, bsdf) {
+  return bsdf * (f0 + (1 - f0) * (1 - abs(VdotH))^5)
+}
+```
+
+For the dielectric BRDF a diffuse component `base` and a specular component `layer` are combined via `fresnel_mix(ior, base, layer)`. The `f0` color is now derived from the index of refraction `ior`.
+
+```
+function fresnel_mix(ior, base, layer) {
+  f0 = ((1-ior)/(1+ior))^2
+  fr = f0 + (1 - f0)*(1 - abs(VdotH))^5
+  return mix(base, layer, fr)
+}
+```
+
+### Metal BRDF and Dielectric BRDF
+
+Applying the functions we arrive at the metal BRDF and dielectric BRDF:
```

Comment:
My understanding is that you are trying to build a bridge here between the old Appendix B content and this new formalization.  I think we are missing some details in this translation of the new representation of the brdf from the old representation.  For example, it's not clear here that 0.04 is the f0 when ior is 1.5.  Do we need to elaborate a bit more on why this section is here, and expand a bit more on how we arrive at the metal_brdf and dielectric_brdf?

Reply:
My idea was to first describe the material as a normative node graph (Section "Complete Model"), then give a non-normative (sample) implementation for the nodes (Sections "Specular BRDF", "Diffuse BRDF", "Fresnel"), and finally substitute all nodes in the graph for their implementations. This is how we arrive at the old (shader code) representation. I added some notes to the beginning of the "Metal BRDF and Dielectric BRDF" section to make the transition more smooth (hopefully). Does that help?